cupy/core/_routines_sorting.pyx

```diff
@@ -69,8 +69,7 @@ cdef ndarray _ndarray_argsort(ndarray self, axis):
                            'reinstall CuPy after uninstalling it.')
 
     if ndim == 0:
-        raise ValueError('Sorting arrays with the rank of zero is not '
-                         'supported')  # as numpy.argsort() raises
+        return cupy.zeros((1,), dtype=cupy.int64)
```

Comment:
The fix is beyond `cat:test`.

- `cupy.ones(()).argsort(axis=2)` should raise `AxisError`.  How about using `atleast_1d`?
- `argmin(axis=0)` and `argmax(axis=0)` have the same issue.

Reply:
Fixed!